viewer/packages/rendering/src/glsl/sector/primitives/generalCylinder.vert

```diff
@@ -67,10 +72,29 @@ void main() {
     vec3 left = normalize(cross(objectToCameraModelSpace, lDir));
     vec3 up = normalize(cross(left, lDir));
 
-    vec3 localBillboardPosition = center + mat3(halfHeight * lDir, a_radius * left, a_radius * up) * position;
+    mat3 billboardWorldRotation = mat3(lDir, left, up);
+    vec3 cylinderAxisScales = vec3(halfHeight, a_radius, a_radius);
+    mat3 inverseBillboardWorldRotation = transpose(billboardWorldRotation);
+    vec3 cameraPosInCylinderSpace = inverseBillboardWorldRotation * (rayOrigin - center);
+
+    mat3 billboardWorldScaleRotation = mat3(halfHeight * lDir, a_radius * left, a_radius * up);
+
+    vec3 localBillboardPosition = center + billboardWorldScaleRotation * position;
     vec3 viewBillboardPosition = mul3(modelToView, localBillboardPosition);
 
-    gl_Position = projectionMatrix * vec4(viewBillboardPosition, 1.0 );
+    float near = projectionMatrix[3][2] / (projectionMatrix[2][2] - 1.0);
```

Comment:
Try using gl_DepthRangeParameters here to fetch near plane (simplify!). https://www.khronos.org/files/opengles3-quick-reference-card.pdf

Reply:
We've discussed and concluded that `gl_DepthRangeParameters` is a dumb uniform